parsec/core/fs/workspacefs/workspacefile.py

```diff
@@ -0,0 +1,159 @@
+# Parsec Cloud (https://parsec.cloud) Copyright (c) AGPLv3 2019 Scille SAS
+
+import os
+import re
+from parsec.core.fs.exceptions import FSUnsupportedOperation, FSOffsetError
+
+
+class WorkspaceFile:
+    """methods of this object should act like trio async file :
+    "https://github.com/python-trio/trio/blob/master/trio/_file_io.py"
+    using transactions"""
```

Comment:
Also it would be nice to document how `mode` works here, especially since `b` is ignored as all the methods deal with bytes.